sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/autorest.md

```diff
@@ -17,6 +17,7 @@ sample-gen:
 skip-csproj: true
 modelerfour:
   flatten-payloads: false
+  lenient-model-deduplication: true
```

Comment:
This flag is deprecated. Do you still want to use it? https://github.com/Azure/autorest/blob/fe71a68f5f65dcb48340a6bb350e9f583b838571/packages/extensions/modelerfour/readme.md?plain=1#L49-L58

Reply:
Without the modelerfour flag we are getting the error below. I've submitted a Swagger PR a while back to fix the duplicated Schema "Resource" but I am still getting this error when running dotnet build /t:GenerateCode command, thus I had to add the tag to be able to generate the code. Any suggestions to resolve this ? 

**PR with the fix for this issue:** https://github.com/Azure/azure-rest-api-specs/pull/24425

**Error when Generating Code without modelerfour tag:** 
![image](https://github.com/Azure/azure-sdk-for-net/assets/25797422/dd70c733-bcf0-4989-80de-65723a3db897)

Reply:
@emmeliaAra There is a `DedicatedHsmResource` which looks like a `Resource` or `TrackedResource`: https://github.com/Azure/azure-rest-api-specs/blob/9a3161dbc683120d907689209a6eebd450af8c3d/specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2021-11-30/dedicatedhsm.json#L807

Also note that there is a [custom `SystemData`](https://github.com/Azure/azure-rest-api-specs/blob/9a3161dbc683120d907689209a6eebd450af8c3d/specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2021-11-30/dedicatedhsm.json#L858) which looks like the one in `common-types`: https://github.com/Azure/azure-rest-api-specs/blob/c1ba9df47907f9012ae14ca4616aed9e5665f9e5/specification/common-types/resource-management/v5/types.json#L502

Better to use the common types as many as possible.

Reply:
Hi, @archerzz - Thanks for pointing out the issue. The issue comes from the definition in DHSM swagger and the new version we are releasing is for Cloud HSM. We are planning to release a new version for both services when we move CHSM to stable folder and I will make sure to make this change in the next PR, if we can go ahead with this one as it is.

Reply:
@emmeliaAra gotcha